### PR TITLE
Migrate BasicEditAndContinue integration tests from Tao to .NET

### PR DESCRIPTION
**Pull request by ivanbasov**
Fixes (almost) #16689 (also will need to remove BasicEditAndContinue.xml at another repository).

Tests only. No need for ask mode approval.

Tagging @dotnet/roslyn-ide for review

### CODE REVIEW

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Debugger_InProc.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+        public void StepOver(bool waitForBreakOrEnd) => _debugger.StepOver(waitForBreakOrEnd);
+
+        public void Stop(bool waitForDesignMode) => _debugger.Stop(WaitForDesignMode: waitForDesignMode);
+
+        public void SetNextStatement() => _debugger.SetNextStatement();
+
+        public void ExecuteStatement(string statement) => _debugger.ExecuteStatement(statement);
+
+        public void CheckExpression(string expression, string expectedType, string expectedValue)
```
Comment (jmarolf) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Debugger_InProc.cs]: >CheckExpression [](start = 20, length = 15)

Why is this needed? can't this just be an assert in the test?

Comment (ivanbasov) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Debugger_InProc.cs]: Thank you! Agree. Moved asserts from InProc to OutProc.
